department-of-veterans-affairs/vets-website
# MHV-48184 Implement-Front-End-Data-Gap-Fixes-For-Health-Conditions

## Summary
I have enhanced reducer/conditions.js by incorporating new functions: extractLocation, extractProvider, and extractProviderNote. These additions ensure that the frontend components receive accurate data. Furthermore, I have developed corresponding tests to thoroughly evaluate these functions' functionality.

## Related issue(s)
[MHV-48184](https://jira.devops.va.gov/browse/MHV-48184) - Implement front end data gap fixes for Health Conditions

Implement front end data gap fixes for Health Conditions

Ensure that the frontend is properly mapped to the correct FHIR fields for this domain. Implement in the following locations:

web
print
pdf
txt

User Story: As a Medical Records User, I want to see the correct data in the associated fields.

Feature Flag Y/N ? N

DataDog/Analytics Y/N ?
Manual Testing Y/N ? Y-Sanjay
Automated Testing Y/N ? N
Accessibility Testing Y/N ? N
UCD Validation Y/N N

Acceptance Criteria:
AC1 Make sure all the fields listed in the comments are mapped appropriately
AC2 All locations have been mapped correctly-web, print, pdf and txt
AC3 Don't access the contained[] resources directly – do a lookup using the extractContainedResource function.
AC4
AC5

Links to UCD:
Sketch Link:
User Flow:
Mobile:
Desktop:
Other Design Notes:

Architectural Notes: Always check for the latest designs in Sketch!

## Testing done
Conducted UI testing to ensure the expected behavior. I updated existing unit test for new components.

## Screenshots

![Screenshot 2024-03-20 at 10 48 20 AM (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/019ac203-51a3-467b-97f9-211ab50c7e45)


## What areas of the site does it impact?

The modification affects the conditionsList section of the Medical Records page.

## Acceptance criteria
AC1 Make sure all the fields listed in the comments are mapped appropriately
AC2 All locations have been mapped correctly-web, print, pdf and txt
AC3 Don't access the contained[] resources directly – do a lookup using the extractContainedResource function.
AC4
AC5

### Quality Assurance & Testing


### Authentication
